### ISSUE TITLE
Don't output full stack trace if cannot connect to jboss

### ISSUE THREAD

**Issue by jewzaam** [bserdar/jcliff#65]
If running jcliff from puppet each line of text in the stack trace is added to the puppet output, which in slow puppet environments is excruciatingly slow.

**Comment by jewzaam** [comment]
Logging in jcliff goes to stdout.  Meaning no other place to put output of an error without broader changes.

@bserdar any objection to start logging to a file (/var/log/jcliff.log probably) and report only top level error and info to stdout?

**Comment by bserdar** [comment]
You can use --output=/var/log/jcliff.log

**Comment by jewzaam** [comment]
Stack trace is written to stdout even when output dir is specified.

**Comment by bserdar** [comment]
What exactly is printed? As far as I can see, all errors are written to the
log.

**Comment by jewzaam** [comment]
```
# jcliff --controller=localhost:12345 --redeploy /etc/redhat/lightblue/web-access-log.conf --output=/tmp/jcliff.log
Setting ruledir to:/usr/share/jcliff-2.9.11/rules
Reading properties from file: /usr/share/jcliff-2.9.11/rules/rules
Reading properties from file: /usr/share/jcliff-2.9.11/rules/system-properties
Reading properties from file: /usr/share/jcliff-2.9.11/rules/path
Reading properties from file: /usr/share/jcliff-2.9.11/rules/ee
Reading properties from file: /usr/share/jcliff-2.9.11/rules/jmx
Reading properties from file: /usr/share/jcliff-2.9.11/rules/naming
Reading properties from file: /usr/share/jcliff-2.9.11/rules/threads
Reading properties from file: /usr/share/jcliff-2.9.11/rules/logging
Reading properties from file: /usr/share/jcliff-2.9.11/rules/jdbc-driver
Reading properties from file: /usr/share/jcliff-2.9.11/rules/security
Reading properties from file: /usr/share/jcliff-2.9.11/rules/datasource
Reading properties from file: /usr/share/jcliff-2.9.11/rules/xadatasource
Reading properties from file: /usr/share/jcliff-2.9.11/rules/standard-sockets
Reading properties from file: /usr/share/jcliff-2.9.11/rules/security-realms
Reading properties from file: /usr/share/jcliff-2.9.11/rules/web
Reading properties from file: /usr/share/jcliff-2.9.11/rules/webservices
Reading properties from file: /usr/share/jcliff-2.9.11/rules/infinispan
Reading properties from file: /usr/share/jcliff-2.9.11/rules/scanner
Reading properties from file: /usr/share/jcliff-2.9.11/rules/subsystem
Reading properties from file: /usr/share/jcliff-2.9.11/rules/remoting
Reading properties from file: /usr/share/jcliff-2.9.11/rules/osgi
Reading properties from file: /usr/share/jcliff-2.9.11/rules/messaging
Reading properties from file: /usr/share/jcliff-2.9.11/rules/mail
Reading properties from file: /usr/share/jcliff-2.9.11/rules/ejb3
Reading properties from file: /usr/share/jcliff-2.9.11/rules/resource-adapter
Reading properties from file: /usr/share/jcliff-2.9.11/rules/transactions
Reading properties from file: /usr/share/jcliff-2.9.11/rules/undertow
Reading properties from file: /usr/share/jcliff-2.9.11/rules/teiid
Exception in thread "main" java.lang.RuntimeException: java.lang.RuntimeException: org.jboss.as.cli.CliInitializationException: Failed to connect to the controller
        at org.jboss.as.cli.impl.CliLauncher.initCommandContext(CliLauncher.java:299)
        at org.jboss.as.cli.impl.CliLauncher.main(CliLauncher.java:259)
        at org.jboss.as.cli.CommandLineMain.main(CommandLineMain.java:45)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:497)
        at org.jboss.modules.Module.run(Module.java:312)
        at org.jboss.modules.Main.main(Main.java:473)
Caused by: org.jboss.as.cli.CommandLineException: The controller is not available at localhost:12345
        at org.jboss.as.cli.impl.CommandContextImpl.tryConnection(CommandContextImpl.java:1058)
        at org.jboss.as.cli.impl.CommandContextImpl.connectController(CommandContextImpl.java:888)
        at org.jboss.as.cli.impl.CommandContextImpl.connectController(CommandContextImpl.java:864)
        at org.jboss.as.cli.impl.CliLauncher.initCommandContext(CliLauncher.java:297)
        ... 8 more
Caused by: java.io.IOException: java.net.ConnectException: JBAS012174: Could not connect to remote://localhost:12345. The connection failed
        at org.jboss.as.controller.client.impl.AbstractModelControllerClient.executeForResult(AbstractModelControllerClient.java:149)
        at org.jboss.as.controller.client.impl.AbstractModelControllerClient.execute(AbstractModelControllerClient.java:75)
        at org.jboss.as.cli.impl.CommandContextImpl.tryConnection(CommandContextImpl.java:1036)
        ... 11 more
Caused by: java.net.ConnectException: JBAS012174: Could not connect to remote://localhost:12345. The connection failed
        at org.jboss.as.protocol.ProtocolConnectionUtils.connectSync(ProtocolConnectionUtils.java:133)
        at org.jboss.as.protocol.ProtocolConnectionManager$EstablishingConnection.connect(ProtocolConnectionManager.java:256)
        at org.jboss.as.protocol.ProtocolConnectionManager.connect(ProtocolConnectionManager.java:70)
        at org.jboss.as.protocol.mgmt.FutureManagementChannel$Establishing.getChannel(FutureManagementChannel.java:208)
        at org.jboss.as.cli.impl.CLIModelControllerClient.getOrCreateChannel(CLIModelControllerClient.java:169)
        at org.jboss.as.cli.impl.CLIModelControllerClient$2.getChannel(CLIModelControllerClient.java:129)
        at org.jboss.as.protocol.mgmt.ManagementChannelHandler.executeRequest(ManagementChannelHandler.java:123)
        at org.jboss.as.protocol.mgmt.ManagementChannelHandler.executeRequest(ManagementChannelHandler.java:98)
        at org.jboss.as.controller.client.impl.AbstractModelControllerClient.executeRequest(AbstractModelControllerClient.java:263)
        at org.jboss.as.controller.client.impl.AbstractModelControllerClient.execute(AbstractModelControllerClient.java:168)
        at org.jboss.as.controller.client.impl.AbstractModelControllerClient.executeForResult(AbstractModelControllerClient.java:147)
        ... 13 more
Caused by: java.net.ConnectException: Connection refused
        at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
        at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:717)
        at org.xnio.nio.NioXnioWorker$1.handleEvent(NioXnioWorker.java:329)
        at org.xnio.nio.NioXnioWorker$1.handleEvent(NioXnioWorker.java:325)
        at org.xnio.ChannelListeners.invokeChannelListener(ChannelListeners.java:72)
        at org.xnio.nio.NioHandle.run(NioHandle.java:90)
        at org.xnio.nio.WorkerThread.run(WorkerThread.java:198)
        at ...asynchronous invocation...(Unknown Source)
        at org.jboss.remoting3.EndpointImpl.doConnect(EndpointImpl.java:286)
        at org.jboss.remoting3.EndpointImpl.doConnect(EndpointImpl.java:267)
        at org.jboss.remoting3.EndpointImpl.connect(EndpointImpl.java:365)
        at org.jboss.remoting3.EndpointImpl.connect(EndpointImpl.java:353)
        at org.jboss.as.protocol.ProtocolConnectionUtils.connect(ProtocolConnectionUtils.java:84)
        at org.jboss.as.protocol.ProtocolConnectionUtils.connectSync(ProtocolConnectionUtils.java:103)
        ... 23 more
        at com.redhat.jcliff.Cli.run(Cli.java:183)
        at com.redhat.jcliff.Cli.run(Cli.java:60)
        at com.redhat.jcliff.Ctx.runcmd(Ctx.java:81)
        at com.redhat.jcliff.Main.refreshServerNode(Main.java:390)
        at com.redhat.jcliff.Main.main(Main.java:212)
Caused by: java.lang.RuntimeException: org.jboss.as.cli.CliInitializationException: Failed to connect to the controller
        at org.jboss.as.cli.impl.CliLauncher.initCommandContext(CliLauncher.java:299)
        at org.jboss.as.cli.impl.CliLauncher.main(CliLauncher.java:259)
        at org.jboss.as.cli.CommandLineMain.main(CommandLineMain.java:45)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:497)
        at org.jboss.modules.Module.run(Module.java:312)
        at org.jboss.modules.Main.main(Main.java:473)
Caused by: org.jboss.as.cli.CommandLineException: The controller is not available at localhost:12345
        at org.jboss.as.cli.impl.CommandContextImpl.tryConnection(CommandContextImpl.java:1058)
        at org.jboss.as.cli.impl.CommandContextImpl.connectController(CommandContextImpl.java:888)
        at org.jboss.as.cli.impl.CommandContextImpl.connectController(CommandContextImpl.java:864)
        at org.jboss.as.cli.impl.CliLauncher.initCommandContext(CliLauncher.java:297)
        ... 8 more
Caused by: java.io.IOException: java.net.ConnectException: JBAS012174: Could not connect to remote://localhost:12345. The connection failed
        at org.jboss.as.controller.client.impl.AbstractModelControllerClient.executeForResult(AbstractModelControllerClient.java:149)
        at org.jboss.as.controller.client.impl.AbstractModelControllerClient.execute(AbstractModelControllerClient.java:75)
        at org.jboss.as.cli.impl.CommandContextImpl.tryConnection(CommandContextImpl.java:1036)
        ... 11 more
Caused by: java.net.ConnectException: JBAS012174: Could not connect to remote://localhost:12345. The connection failed
        at org.jboss.as.protocol.ProtocolConnectionUtils.connectSync(ProtocolConnectionUtils.java:133)
        at org.jboss.as.protocol.ProtocolConnectionManager$EstablishingConnection.connect(ProtocolConnectionManager.java:256)
        at org.jboss.as.protocol.ProtocolConnectionManager.connect(ProtocolConnectionManager.java:70)
        at org.jboss.as.protocol.mgmt.FutureManagementChannel$Establishing.getChannel(FutureManagementChannel.java:208)
        at org.jboss.as.cli.impl.CLIModelControllerClient.getOrCreateChannel(CLIModelControllerClient.java:169)
        at org.jboss.as.cli.impl.CLIModelControllerClient$2.getChannel(CLIModelControllerClient.java:129)
        at org.jboss.as.protocol.mgmt.ManagementChannelHandler.executeRequest(ManagementChannelHandler.java:123)
        at org.jboss.as.protocol.mgmt.ManagementChannelHandler.executeRequest(ManagementChannelHandler.java:98)
        at org.jboss.as.controller.client.impl.AbstractModelControllerClient.executeRequest(AbstractModelControllerClient.java:263)
        at org.jboss.as.controller.client.impl.AbstractModelControllerClient.execute(AbstractModelControllerClient.java:168)
        at org.jboss.as.controller.client.impl.AbstractModelControllerClient.executeForResult(AbstractModelControllerClient.java:147)
        ... 13 more
Caused by: java.net.ConnectException: Connection refused
        at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
        at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:717)
        at org.xnio.nio.NioXnioWorker$1.handleEvent(NioXnioWorker.java:329)
        at org.xnio.nio.NioXnioWorker$1.handleEvent(NioXnioWorker.java:325)
        at org.xnio.ChannelListeners.invokeChannelListener(ChannelListeners.java:72)
        at org.xnio.nio.NioHandle.run(NioHandle.java:90)
        at org.xnio.nio.WorkerThread.run(WorkerThread.java:198)
        at ...asynchronous invocation...(Unknown Source)
        at org.jboss.remoting3.EndpointImpl.doConnect(EndpointImpl.java:286)
        at org.jboss.remoting3.EndpointImpl.doConnect(EndpointImpl.java:267)
        at org.jboss.remoting3.EndpointImpl.connect(EndpointImpl.java:365)
        at org.jboss.remoting3.EndpointImpl.connect(EndpointImpl.java:353)
        at org.jboss.as.protocol.ProtocolConnectionUtils.connect(ProtocolConnectionUtils.java:84)
        at org.jboss.as.protocol.ProtocolConnectionUtils.connectSync(ProtocolConnectionUtils.java:103)
        ... 23 more
        at com.redhat.jcliff.Cli.run(Cli.java:174)
        ... 4 more
```

**Comment by bserdar** [comment]
Fixed in 2.11.2
